tests/python_client/testcases/test_index.py

```diff
@@ -1310,3 +1315,188 @@ def test_create_index_callback(self, connect, collection, get_simple_index):
         res = future.result()
         # TODO:
         log.info(res)
+
+
+class  TestIndexString(TestcaseBase):
+    """
+    ******************************************************************
+      The following cases are used to test create index about string 
+    ******************************************************************
+    """   
+    @pytest.mark.tags(CaseLabel.L1)
+    def test_create_index_with_string_field(self):
+        """
+        target: test create index with string field is not primary
+        method: 1.create collection and insert data 
+                2.only create an index with string field is not primary
+        expected: create index successfully
+        """
+        c_name = cf.gen_unique_str(prefix)
+        collection_w = self.init_collection_wrap(name=c_name)
+        data = cf.gen_default_list_data()
+        collection_w.insert(data=data)
+        index, _ = self.index_wrap.init_index(collection_w.collection, default_string_field_name, default_string_index_params)
+        cf.assert_equal_index(index, collection_w.collection.indexes[0])
```

Comment:
I think "collection_w.collection.indexes" should be replaced by "collection_w.indexes". And all of these code should be modified.